packages/yarn-plugin-allow-scripts/src/index.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  loadAllPackageConfigurations,
+  printMissingPoliciesIfAny,
+} from '@lavamoat/allow-scripts'
+import { Plugin, type Project, type Locator } from '@yarnpkg/core'
+import { dirname } from 'node:path'
+
+// Disable lifecycle scripts for transitive git deps with a yarn classic lockfile
+if (typeof process.env['YARN_IGNORE_SCRIPTS'] === 'undefined') {
+  process.env['YARN_IGNORE_SCRIPTS'] = 'true'
+}
+
+interface ExtendedProject extends Project {
+  allowScriptsConfig?: Map<string, boolean>
+}
+
+// Yarn inserts spurious / for absolute paths on Windows
+const yarnCwdToPath = (cwd: string): string =>
+  // eslint-disable-next-line no-useless-escape
+  process.platform === 'win32' ? cwd.replace(/^[\/]/, '') : cwd
+
+const isPackageScriptAllowed = async (
+  project: ExtendedProject,
+  npm_package_json: string
+): Promise<boolean> => {
+  if (!project.allowScriptsConfig) {
+    project.allowScriptsConfig = new Map<string, boolean>()
+  }
+  const {
+    canonicalNamesByPath,
+    configs: { lifecycle },
+  } = await loadAllPackageConfigurations({
+    rootDir: yarnCwdToPath(project.cwd),
+  })
+  const scriptDir = dirname(npm_package_json)
+  const packageName = canonicalNamesByPath.get(scriptDir)
+  if (!packageName) {
+    process.stderr.write(
+      `FATAL ERROR: allow-scripts blocked execution of unexpected package script. ${JSON.stringify(
+        {
+          package: packageName,
+          manifestPath: npm_package_json,
+        }
+      )}`
+    )
+    process.exit(1)
+  }
+  if (lifecycle.disallowedPatterns.includes(packageName)) {
+    project.allowScriptsConfig.set(packageName, false)
+    return false
+  }
+  if (lifecycle.allowedPatterns.includes(packageName)) {
+    project.allowScriptsConfig.set(packageName, true)
+    return true
+  }
+  return null
+}
+
+const plugin: Plugin = {
+  hooks: {
+    wrapScriptExecution: (
+      executor: () => Promise<number>,
+      project: ExtendedProject,
+      _locator: Locator,
+      npm_lifecycle_event: string,
+      { env: { npm_package_json, npm_package_name } }: { env: any }
+    ): Promise<() => Promise<number>> => {
+      return new Promise((resolve, _reject) =>
+        isPackageScriptAllowed(project, npm_package_json).then((isAllowed) => {
+          if (!isAllowed) {
+            console.error(
+              isAllowed === null
+                ? `  allow-scripts blocking execution of unconfigured package script. ${JSON.stringify([npm_package_name, npm_lifecycle_event, npm_package_json])}`
+                : `  allow-scripts blocking execution of disallowed package script. ${JSON.stringify([npm_package_name, npm_lifecycle_event, npm_package_json])}`
+            )
+            return resolve(() => Promise.resolve(0))
```

Comment:
(could) use a named function here to indicate what the function is.
```js
const noopScriptExecutor = async ()=>{}
```

```suggestion
            return resolve(noopScriptExecutor)
```

Reply:
I'd add if it'd been used in more then one place, or been less trivial.